### PR TITLE
[SPARK-332] Adds CNI support

### PR DESCRIPTION
**Pull request by mgummelt**
cc @susanxhuynh for review

1) Updates spark-env.sh to use DNS only when it's available (because CNI requires it)
2) Adds a CNI integration test
3) Refactors the tests to use py.test rather than python directly (same as other infinity frameworks)

### CODE REVIEW

File: tests/test.py [hunk 1]
```diff
@@ -13,50 +13,57 @@
...
-        os.environ['S3_BUCKET'],
-        os.environ['S3_PREFIX'],
-        basename)
+def test_cni():
+    SPARK_EXAMPLES="http://downloads.mesosphere.com/spark/assets/spark-examples_2.11-2.0.1.jar"
+    _run_tests(SPARK_EXAMPLES,
+               "",
+               "Pi is roughly 3",
+               {"--class": "org.apache.spark.examples.SparkPi"},
+               {"spark.mesos.network.name": "dcos"})
```
Comment (susanxhuynh) [inline tests/test.py]: Is this the name of the network in the launched cluster?

**Comment by mgummelt** [comment]
retest this please

File: tests/test.py (hunk 1 above)
Comment (susanxhuynh) [inline tests/test.py]: Okay. LGTM if the tests pass.

**Comment by mgummelt** [comment]
retest this please

**Pull request merged**